.changeset/chilly-badgers-push.md

```diff
@@ -0,0 +1,29 @@
+---
+'@astrojs/solid-js': major
+---
+
+Render SolidJS components using [renderToStringAsync](https://www.solidjs.com/docs/latest#rendertostringasync).
+
+This changes the renderer of SolidJS components from renderToString to renderToStringAsync. It also injected the full SolidJS hydration script generated by `generateHydrationScript`, to help make sure that Suspense, ErrorBoundary and similar features can be hydrated correctly.
+
+The server render phase will now wait for Suspense boundaries to resolve instead of immediately rendering the [Suspense](https://www.solidjs.com/docs/latest#suspense) fallback.
+
+If your SolidJS component uses APIs such as [lazy](https://www.solidjs.com/docs/latest#lazy) or [createResource](https://www.solidjs.com/docs/latest#createresource), these functions may now be used on the server side.
+
+This increases the flexibility of the SolidJS integration. Server-only SolidJS components may now call async functions directly using the `createResource` API, like loading data from another API or using async Astro server functions like `getImage()`. It is very unlikely that a server only component would have used the Suspense feature until now, so this should not be a breaking change for server-only components.
+
+This could be a breaking change for components that meet the following conditions:
+
+- The component uses Suspense APIs like `Suspense`, `lazy` or `createResource`, and
+- The component is mounted using a *hydrating* directive:
+  - `client:load`
+  - `client:idle`
+  - `client:visible`
+  - `client:media`
```

Comment:
```suggestion
This increases the flexibility of the SolidJS integration. Server-only SolidJS components may now call async functions directly using the `createResource` API, like loading data from another API or using async Astro server functions like `getImage()`.

It is very unlikely that a server only component would have used the Suspense feature until now, so this should not be a breaking change for server-only components.

However, this could be a breaking change for components that meet the following conditions:

- The component uses Suspense APIs like `Suspense`, `lazy` or `createResource`, and
- The component is mounted using a *hydrating* directive:
  - `client:load`
  - `client:idle`
  - `client:visible`
  - `client:media`
```
I always like to call out more attention when we're talking about breaking changes, and so that first sentence probably shouldn't be tacked on to the end of an unrelated paragraph. The "however" gives that extra bit of warning, and doesn't make it sound like you're contradicting yourself! :smile:

Reply:
Addressed in https://github.com/withastro/astro/pull/6791/commits/5a810a9b73463d93b3b7eeee92ea1dd1ca9f7111